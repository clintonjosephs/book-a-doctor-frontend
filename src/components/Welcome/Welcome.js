import React from 'react';
import AnimateWrapper from '../AnimateWrapper';

const Welcome = () => (
  <AnimateWrapper
    initial={{ x: 300, opacity: 1 }}
    animate={{ x: 0, opacity: 1 }}
    exit={{ x: 300, opacity: 0 }}
    transition={{ duration: 0.8 }}
  >
    <div>Available Doctors</div>
  </AnimateWrapper>
);

export default Welcome;
