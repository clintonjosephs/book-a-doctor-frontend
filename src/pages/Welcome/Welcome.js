import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchAllDoctors } from '../../redux/bookDoctor/doctorThunks';
import { updateCarouselState } from '../../redux/bookDoctor/doctorActions';
import CarouselItem from '../../components/Doctors/Carousel/CarouselItem';
import AnimateWrapper from '../../components/AnimateWrapper';
import styles from './Welcome.module.css';

const Welcome = () => {
  const [current, setCurrent] = useState(1);

  const dispatch = useDispatch();
  const {
    currentCarouselState,
    loading,
    doctorsChunked,
  } = useSelector((state) => state.bookDoctorReducer);

  useEffect(() => {
    dispatch(fetchAllDoctors());
  }, []);

  if (doctorsChunked.length === 0 && loading) {
    return <h2>Loading</h2>;
  }

  if (doctorsChunked.length === 0 && !loading) {
    return <h2>No doctors available at the moment</h2>;
  }

  const { length } = doctorsChunked;

  const nextSlide = () => {
    setCurrent(current === length - 1 ? 0 : current + 1);
    dispatch(updateCarouselState(current));
  };

  const prevSlide = () => {
    setCurrent(current === 0 ? length - 1 : current - 1);
    dispatch(updateCarouselState(current));
  };

  return (
    <AnimateWrapper
      initial={{ y: 1000, opacity: 1 }}
      animate={{ y: 0, opacity: 1 }}
      exit={{ y: 1000, opacity: 0 }}
      transition={{ duration: 0.8 }}
    >
      <div className={styles.top}>
        <span><b>AVAILABLE DOCTORS</b></span>
        <p>Please select a doctor</p>
      </div>
      <section className={styles.doctorsCarousel}>
        <button type="button" onClick={prevSlide} className={`${styles.prev} ${styles.direction}`}>
          <i className="fa-solid fa-caret-left" />
        </button>
        <ul className={styles.carouselWrapper}>
          { currentCarouselState && currentCarouselState.map(({
            id, name, imageUrl, description,
          }) => (
            <CarouselItem
              key={id}
              name={name}
              imageUrl={imageUrl}
              id={id}
              description={description}
            />
          )) }
        </ul>
        <button type="button" onClick={nextSlide} className={`${styles.next} ${styles.direction}`}>
          <i className="fa-solid fa-caret-right" />
        </button>
      </section>

    </AnimateWrapper>
  );
};

export default Welcome;
