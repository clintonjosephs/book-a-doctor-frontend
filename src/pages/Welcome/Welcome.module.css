.doctorsCarousel {
  margin-left: 15rem;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  position: relative;
  z-index: -100;
}

.noDoctors {
  margin-top: 15rem;
}

.next {
  position: absolute;
  top: 50%;
  right: 32px;
  z-index: 10;
  border-top-left-radius: 2rem;
  border-bottom-left-radius: 2rem;
}

.prev {
  position: absolute;
  top: 50%;
  left: 32px;
  z-index: 10;
  border-top-right-radius: 2rem;
  border-bottom-right-radius: 2rem;
}

.direction {
  padding: 0.75rem 1rem;
  background-color: var(--primary-color);
  border: none;
  color: var(--white);
  font-size: 1rem;
  cursor: pointer;
  width: 4rem;
  height: 3rem;
}

.top {
  text-align: center;
  margin-top: 2rem;
  margin-bottom: 3rem;
  margin-left: 15rem;
}

.top span {
  font-size: 2rem;
}

.top p {
  color: var(--dull);
  margin-top: 0.7rem;
  font-size: large;
}

.slideImg {
  border-radius: 50%;
  align-self: center;
  width: 280px;
  height: 280px;
}

.docName {
  font-size: 1.2rem;
  font-weight: 500;
  text-align: center;
  margin-top: 1rem;
}

.docDescription {
  font-size: 16px;
  font-weight: 500;
  text-align: center;
  line-height: 1.5rem;
  color: var(--dull);
  margin-left: 2rem;
  margin-right: 2rem;
  width: 18rem;
}

.carouselItem {
  display: flex;
  flex-direction: column;
  justify-items: center;
}

.line {
  width: 7rem;
  border: dotted 1px var(--primary-color);
  align-self: center;
  margin: 2rem;
}

.doctorsCarousel ul {
  display: flex;
  flex-direction: row;
  list-style: none;
  padding: 0;
  margin: 0;
}

.doctorsCarousel ul li {
  cursor: pointer;
}

.social {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.social img {
  width: 2rem;
  height: 2rem;
  margin: 10px;
}

.animate {
  animation: createBox 2s;
}

@keyframes createBox {
  from {
    transform: scale(0.7);
  }

  to {
    transform: scale(1);
  }
}

@media (max-width: 768px) {
  .doctorsCarousel {
    margin: 1rem;
    height: 100%;
  }

  .doctorsCarousel ul {
    flex-direction: column;
  }

  .doctorsCarousel ul li {
    margin-bottom: 2rem;
    border-bottom: solid var(--dull);
  }

  .top {
    margin-left: 2.5rem;
    margin-right: 0;
  }

  .top span {
    font-size: 1.5rem;
  }

  .next {
    top: 100%;
  }

  .prev {
    top: 100%;
  }
}
