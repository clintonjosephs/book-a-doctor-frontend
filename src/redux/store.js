import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import bookDoctorReducer from './bookDoctor/doctorActions';
import userReducer from './user/userActions';
import reservationReducers from './reservation/reservationActions';

const reducer = combineReducers({
  bookDoctorReducer,
  userReducer,
  reservationReducers,
});

// creating a store
const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
