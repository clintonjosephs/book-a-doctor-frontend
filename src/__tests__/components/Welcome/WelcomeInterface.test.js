import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/store';
import Welcome from '../../../pages/Welcome/Welcome';

const WelcomeProvider = () => (
  <Provider store={store}>
    <Router>
      <Welcome />
    </Router>
  </Provider>
);

describe('check login components', () => {
  it('matches actual design', () => {
    const component = renderer.create(<WelcomeProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
