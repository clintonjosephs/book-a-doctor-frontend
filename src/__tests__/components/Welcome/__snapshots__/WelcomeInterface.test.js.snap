// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check login components matches actual design 1`] = `
<div
  className="top"
>
  <span>
    <b>
      AVAILABLE DOCTORS
    </b>
  </span>
  <p>
    Please select a doctor
  </p>
  <div
    className="noDoctors"
  >
    No Available doctors yet ðŸ˜ž 
  </div>
</div>
`;
